src/executor/executor.cc

```diff
@@ -0,0 +1,60 @@
+#include "executor/executor.hh"
+
+namespace sim {
+
+static void execute_ADD(const Instruction &inst, State &state) {
+    Word RS1 = state.regs.get(inst.RS1);
+    Word RS2 = state.regs.get(inst.RS1);
+    state.regs.set(inst.RD, RS1 + RS2);
+    state.pc += XLEN_IN_BYTES;
+}
+
+static void execute_SUB(const Instruction &inst, State &state) {
+    Word RS1 = state.regs.get(inst.RS1);
+    Word RS2 = state.regs.get(inst.RS1);
+    state.regs.set(inst.RD, RS1 - RS2);
+    state.pc += XLEN_IN_BYTES;
+}
+
+static void execute_MUL(const Instruction &inst, State &state) {
+    Word RS1 = state.regs.get(inst.RS1);
+    Word RS2 = state.regs.get(inst.RS1);
+    state.regs.set(inst.RD, RS1 * RS2);
+    state.pc += XLEN_IN_BYTES;
+}
+
+static void execute_DIV(const Instruction &inst, State &state) {
+    Word RS1 = state.regs.get(inst.RS1);
+    Word RS2 = state.regs.get(inst.RS1);
+    state.regs.set(inst.RD, RS1 / RS2);
+    state.pc += XLEN_IN_BYTES;
+}
+
+static void execute_LW(const Instruction &inst, State &state) {
+    Word RS1 = state.regs.get(inst.RS1);
+    Word offset = inst.imm;
+    state.regs.set(inst.RD, state.mem.load(RS1 + offset));
+    state.pc += XLEN_IN_BYTES;
+}
+
+static void execute_SW(const Instruction &inst, State &state) {
+    Word address = state.regs.get(inst.RS1) + inst.imm;
+    Word RS2 = state.regs.get(inst.RS1);
+    state.mem.store(address, RS2);
+}
```

Comment:
You have forgotten to increase PC. I think PC shouldn't be increased in execute_* functions. It should be handled in higher level. For example in execute method after handler call